src/game/logic/object/weapon.h

```diff
@@ -112,3 +151,179 @@ class WeaponBonusSet : public MemoryPoolObject
 private:
     WeaponBonus m_bonus[WEAPONBONUSCONDITION_COUNT];
 };
+
+struct HistoricWeaponDamageInfo
+{
+    uint32_t frame;
+    Coord3D location;
+};
+
+class WeaponTemplate : public MemoryPoolObject
+{
+    IMPLEMENT_POOL(WeaponTemplate);
+
+public:
+    WeaponTemplate &operator=(WeaponTemplate &rhs);
+
+    bool Has_Next_Template() const { return m_nextTemplate != nullptr; }
+    const Utf8String &Get_Name() const { return m_name; }
+    NameKeyType Get_Name_Key() const { return m_nameKey; }
+    const Utf8String &Get_Projectile_Name() const { return m_projectileObject; }
+    const AudioEventRTS &Get_Fire_Sound() const { return m_fireSound; }
+    uint32_t Get_Damage_Type() const { return m_damageType; }
+
+    // wb: 0x006D6B4E friend?
+    WeaponTemplate *Clear_Next_Template()
+    {
+        auto *ret = m_nextTemplate;
+        m_nextTemplate = nullptr;
+        return ret;
+    }
+    void Clear_Projectile_Name() { m_projectileObject.Clear(); }
+    void Set_Name(const Utf8String &newName);
+    void Set_Next(WeaponTemplate *next) { m_nextTemplate = next; }
+    // wb: 0x006D3E85
+    void Reset() { m_historicDamage.clear(); }
+    void Deal_Damage_Internal(
+        uint32_t sourceID, uint32_t victimID, Coord3D *pos, WeaponBonus &bonus, bool isProjectileDetonation);
+    void PostProcessLoad();
+
+    static void Parse_Scatter_Target(INI *ini, void *formal, void *store, void const *user_data);
+    static void Parse_Weapon_Bonus_Set(INI *ini, void *formal, void *store, void const *user_data);
+    static void Parse_Shot_Delay(INI *ini, void *formal, void *store, void const *user_data);
+    static void Parse_Weapon_Template(INI *ini, void *formal, void *store, const void *user_data);
+    static FieldParse *Get_Weapon_Template_Field_Parse_Table() { return s_fieldParseTable; }
+
+private:
+    WeaponTemplate *m_nextTemplate;
+    Utf8String m_name;
+    NameKeyType m_nameKey;
+    Utf8String m_projectileStreamName;
+    Utf8String m_laserName;
+    Utf8String m_laserBoneName;
+    float m_primaryDamage;
+    float m_primaryDamageRadius;
+    float m_secondaryDamage;
+    float m_secondaryDamageRadius;
+    float m_shockWaveAmount;
+    float m_shockWaveRadius;
+    float m_shockWaveTaperOff;
+    float m_attackRange;
+    float m_minimumAttackRange;
+    float m_requestAssistRange;
+    float m_aimDelta;
+    float m_scatterRadius;
+    float m_scatterTargetScalar;
+    std::vector<Coord2D> m_scatterTarget;
+    DamageType m_damageType;
+    DeathType m_deathType;
+    float m_weaponSpeed;
+    float m_minWeaponSpeed;
+    bool m_scaleWeaponSpeed;
+    float m_weaponRecoil;
+    float m_minTargetPitch;
+    float m_maxTargetPitch;
+    float m_radiusDamageAngle;
+    Utf8String m_projectileObject;
+    uint32_t unk;
+    Utf8String m_fireOCL[VETERANCY_COUNT];
```

Comment:
ok these
not sure if correct
not sure if correct
m_projectileExhausts , later SAGE xsd's have comment naming it
not sure if correct
wrong
m_fireFXs , later SAGE xsd's have comment naming it
need to check for real name

Reply:
Unknowns are ObjectCreationList *[VETERANCY_COUNT], WB handles them at 006D40BD WeaponStore::postProcessLoad

Reply:
Seems its FireOCL and ProjectileDetonationOCL the actual OCLs while earlier ones are just names